pkg/provider/aws/static_unit.go

```diff
@@ -29,15 +28,30 @@ func (a *AWS) expandStaticUnit(dag *core.ResourceGraph, unit *core.StaticUnit) e
 		}
 		createdBuckets[object.Bucket.Id()] = object.Bucket
 	}
-	nBuckets := len(createdBuckets)
-	zap.L().With()
-	if nBuckets > 0 {
-		if nBuckets > 1 {
-			errs.Append(errors.Errorf(`Found too many buckets for unit %s. This is an internal error.`, unit.Id()))
-		}
+	if err := errs.ErrOrNil(); err != nil {
+		return err
+	}
+	switch len(createdBuckets) {
+	case 0:
+		return nil
+	case 1:
 		_, bucket := collectionutil.GetOneEntry(createdBuckets)
 		a.MapResourceDirectlyToConstruct(bucket, unit)
-	}
 
-	return errs.ErrOrNil()
+		cfg := a.Config.GetStaticUnit(unit.ID)
+		if cfg.ContentDeliveryNetwork.Id != "" {
+			distro, err := core.CreateResource[*resources.CloudfrontDistribution](dag, resources.CloudfrontDistributionCreateParams{
+				CdnId:   cfg.ContentDeliveryNetwork.Id,
+				AppName: a.Config.AppName,
+				Refs:    core.AnnotationKeySetOf(unit.AnnotationKey),
+			})
+			if err != nil {
+				return err
+			}
+			dag.AddDependency(distro, bucket)
+		}
+		return nil
+	default:
+		return errors.Errorf(`Found too many buckets for unit %s. This is an internal error.`, unit.Id())
```

Comment:
```suggestion
		var bucketIds []core.ResourceId
		for k := range createdBuckets {
			bucketIds = append(bucketIds, k)
		}
		return errors.Errorf(`Found too many buckets %v for unit %s. This is an internal error.`, bucketIds, unit.Id())
```
nit: just a little more info/context on the error